File: deno/flowrepl/api/InterpreterRequest.ts
export class InterpreterRequest {
    id = -1;
    code: string;
    language = "javascript";
    history = false;

    constructor({
        id = -1,
        code,
        language = "javascript",
        history = false,
    }: {
        id?: number;
        code: string;
        language?: string;
        history?: boolean;
    }) {
        this.id = id;
        this.code = code;
        this.language = language;
        this.history = history;
    }
}

File: kotlin-repl/src/main/kotlin/org/clickprompt/unitserver/repl/api/InterpreterRequest.kt
package org.clickprompt.unitserver.repl.api

import kotlinx.serialization.Serializable

@Serializable
data class InterpreterRequest(
    var id: Int = -1,
    var code: String,
    val language: String = "kotlin",
    val framework: String = SupportedFramewrok.SPRING.toString(),
    val history: Boolean = false,
    var port: Int = 8080
)

enum class SupportedFramewrok {
    SPRING, KTOR;

    companion object {
        fun fromString(value: String): SupportedFramewrok {
            return when (value.lowercase()) {
                "spring" -> SPRING
                "ktor" -> KTOR
                else -> throw IllegalArgumentException("Unknown framework: $value")
            }
        }
    }
}
